analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -19,17 +19,25 @@
  */
 
 using System.IO;
-using Microsoft.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
 using Microsoft.CodeAnalysis.Text;
 
 namespace SonarAnalyzer;
 
 public abstract class SonarAnalysisContextBase
 {
+    private static readonly Lazy<SourceTextValueProvider<bool>> ShouldAnalyzeGeneratedCS = new(() => CreateAnalyzeGeneratedProvider(LanguageNames.CSharp));
+    private static readonly Lazy<SourceTextValueProvider<bool>> ShouldAnalyzeGeneratedVB = new(() => CreateAnalyzeGeneratedProvider(LanguageNames.VisualBasic));
     private static readonly SourceTextValueProvider<ProjectConfigReader> ProjectConfigProvider = new(x => new ProjectConfigReader(x));
+    private static readonly ConditionalWeakTable<Compilation, ImmutableHashSet<string>> UnchangedFilesCache = new();
 
     public abstract bool TryGetValue<TValue>(SourceText text, SourceTextValueProvider<TValue> valueProvider, out TValue value);
 
+    public bool ShouldAnalyze(GeneratedCodeRecognizer generatedCodeRecognizer, SyntaxTree tree, Compilation compilation, AnalyzerOptions options) =>
```

Comment:
Can be protected. Is there any reason to make it public?

Reply:
It needs to be public, there are external invocations to it. 
At least this one https://github.com/SonarSource/sonar-dotnet/blob/e14c642f118958f22fd08841dc42f9aae480366a/analyzers/src/SonarAnalyzer.CSharp/Rules/WcfMissingContractAttribute.cs#L89
And also once `SonarAnalysisContext` will not inherit `SonarAnalysisContextBase`, it will need a public access to it.